### PR TITLE
Cybnon/add modbus and mbus smartmeter reading

### PR DESCRIPTION
**Pull request by Cybnon**

### CODE REVIEW

File: app/dependencies.py [hunk 1]
```diff
@@ -17,11 +18,12 @@ def __init__(self):
...
         self.config_base_path = os.environ.get("CONFIG_PATH") or "/tmp"
         self.smd_topic = os.environ.get("SMD_TOPIC") or "rddl/SMD/#"
-        self.path_to_smart_meter_config = build_config_path(self.config_base_path, FILE_SMART_METER_CONFIG)
         self.path_to_mqtt_config = build_config_path(self.config_base_path, FILE_MQTT_CONFIG)
+        self.path_to_smart_meter_config = build_config_path(self.config_base_path, FILE_SMART_METER_CONFIG)
+        self.path_to_smart_meter_mqtt_config = build_config_path(self.config_base_path, FILE_SMART_METER_MQTT_CONFIG)
         self.trust_wallet_port = os.environ.get("TRUST_WALLET_PORT") or "/dev/ttyACM0"
-        self.notarize_interval = int(os.environ.get("NOTARIZE_INTERVAL") or 60)
-        self.rddl_network_mode = os.environ.get("RDDL_NETWORK_MODE") or "mainnet"
+        self.notarize_interval = int(os.environ.get("NOTARIZE_INTERVAL") or 2)
```
Comment (eckelj) [inline app/dependencies.py:25]: aren't these test values? shouldn't we set them back to 60 and mainnet?

File: app/energy_agent/smart_meter_reader/modbus_reader.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+from pymodbus.client import ModbusSerialClient
+import time
+import binascii
+import re
+from app.helpers.logs import logger, log
+
+# Configuration
+SERIAL_PORT = "/dev/ttyUSB0"
```
Comment (eckelj) [inline app/energy_agent/smart_meter_reader/modbus_reader.py:8]: does this potentially collide with the trust wallet? shouldn't we make this configurable?

File: pyproject.toml [hunk 3]
```diff
@@ -14,7 +14,7 @@ pydantic = "^2.8.2"
 gurux-dlms = "^1.0.150"
 lxml = "^5.2.2"
 cryptography = "^42.0.8"
-fastapi = "^0.111.1"
+fastapi = "^        0.111.1"
```
Comment (eckelj) [inline pyproject.toml:17]: please remove the whitespace

**Review by eckelj — CHANGES_REQUESTED** [review]

File: app/dependencies.py (hunk 1 above)
Comment (Cybnon) [reply app/dependencies.py:25]: You are right i overlooked that one!

File: app/energy_agent/smart_meter_reader/modbus_reader.py (hunk 2 above)
Comment (Cybnon) [reply app/energy_agent/smart_meter_reader/modbus_reader.py]: This is just the default config. 

i removed the constant and added the default parameter to the initalisation of that Reader, 

The actual config would come from the SmartMeterConfig loaded before startup. Than can be defined on the energy agent webpage under additional options smartmeter

File: pyproject.toml (hunk 3 above)
Comment (Cybnon) [reply pyproject.toml]: done

**Review by eckelj — APPROVED** [review]

**Pull request merged**